src/main/java/gregtech/common/items/MetaItem2.java

```diff
@@ -69,7 +70,7 @@ public void registerSubItems() {
         POWER_UNIT_HV = addItem(575, "power_unit.hv") .addComponents(ElectricStats.createElectricItem(1600000L, GTValues.HV)).setMaxStackSize(8);
         JACKHAMMER_BASE = addItem(576, "jackhammer_base").addComponents(ElectricStats.createElectricItem(1600000L, GTValues.HV)).setMaxStackSize(4);
 
-        NANO_SABER = addItem(577, "nano_saber").addComponents(ElectricStats.createElectricItem(4000000L, GTValues.HV)).addComponents(new NanoSaberBehavior(5.0f, 20.0f, 64)).setMaxStackSize(1);
+        NANO_SABER = addItem(577, "nano_saber").addComponents(ElectricStats.createElectricItem(4000000L, GTValues.HV)).addComponents(new NanoSaberBehavior((float)ConfigHolder.nanoSaberConfiguration.nanoSaberBaseDamage, (float)ConfigHolder.nanoSaberConfiguration.nanoSaberDamageBoost, ConfigHolder.nanoSaberConfiguration.energyConsumption)).setMaxStackSize(1);
```

Comment:
Using it here with all that casting does not feel right to me. I think that `NanoSaberBehavior` should have parameter-less constructor and all getting values from config should be done in it.  Also insides of `NanoSaberBehavior` should be changed to use double instead of floats to prevent all that casting.

Reply:
Moved to a parameter-less constructor, with values populated based on the config option. Values additionally moved to doubles to prevent casting.